adoc/chapters/programming_interface.adoc

```diff
@@ -6537,6 +6537,8 @@ capture the [code]#accessor# in a <<host-task>> can only use the accessor for
 interoperability through the [code]#interop_handle#, any other uses result in
 undefined behavior.
 
+An accessor with access target [code]#target::host_task# is accessible on the
+host, within the host task function for which the accessor is declared.
```

Comment:
An accessor in a HT can be used to access device memory (if constructed with `access::target::device`), or it can be used to access host side memory (if constructed with `access::target::host_task`).

But `access::target::host_task` doesn't tell the user where the memory is accessible from, just that it is accessible within the scope of the HT function. Since we can have both device and host accessors within HTs we need to be explicit that the `access::target::host_task` accessor should only be used on host, and not device through some interop method.

Reply:
I see.  I think we can convey this more clearly with a larger change.  I think the problem is that we initially wrote this section without thinking about the interop case, and then we added that in as an afterthought.  It would probably have been better to introduce all three uses up front in the introductory paragraph of this section like so:

> The `accessor` class can be used to access a `buffer` in three different ways.  It can be used to access the buffer's data from within a SYCL kernel function via the device's global memory.  It can also be used to access the buffer's data from within a host task.  Finally, it can be used to get a native backend handle to the buffer from within a host task.  The `AccessTarget` template parameter helps distinguish these three cases as shown in Table 54.
>
> [Table 54]
> `target::device`: Access a buffer's data from within a SYCL kernel function.  Also used to get a native backend handle to the buffer from within a host task.
> `target::host_task`: Access a buffer's data from within a host task.
>
> When an `accessor` is used from within a kernel function, the access target must be `target::device`, otherwise the behavior is undefined.
>
> When an `accessor` is used from within a host task, the use of the accessor must correspond to the access target, otherwise the behavior is undefined.  If the access target is `target::host_task`, the accessor may only be used to access the buffer's data.  If the access target is `target::device`, the accessor may only be used to get a native backend handle for the buffer as described in Section 4.10.2.

Reply:
```quote

Reply:
Sounds good but I think worth also adding "on host" in a few places

> ... It can also be used to access the buffer's data ***on host*** from within a host task.  Finally, it can be used to get a native backend handle to the buffer from within a host task.  The `AccessTarget` template parameter helps distinguish these three cases as shown in Table 54.
> [Table 54]
> `target::device`: Access a buffer's data from within a SYCL kernel function.  Also used to get a native backend handle to the buffer from within a host task.
> `target::host_task`: Access a buffer's data ***on host*** from within a host task.

> ... If the access target is `target::host_task`, the accessor may only be used to access the buffer's data ***on host***.

Reply:
Yes, that seems like a good improvement.  Thanks.